Navigation drawer overhaul

karllindmark opened the pull request:
Currently includes:
- Refactoring of Navigation Drawer implementation
- All previously searched soldiers are now accessible in the dropdown
- LoginActivity now only returns ONE soldier (matched on personaId+platformId)

TODO:
- Decide what to show on HomeFragment
- Make the Navigation Drawer display the current menu selection state properly (maybe change background resource?)
- Update to use v7 version of ActionbarDrawerToggle (+Toolbar so that the navigation drawer slides over it)
- Animations for the menu toggling (when you show/hide the soldiers) ala I/O?

[WIP] @peter-budo

Commit:
[WIP] Navigation Drawer overhaul [SQUASHED]

Currently includes:
+ Refactoring of Navigation Drawer implementation
+ All previously searched soldiers are now accessable in the dropdown

TODO:
- Decide what to show on HomeFragment
- Make LoginActivity only return ONE soldier (and not X) by using personaId+platformId
- Make the Navigation Drawer display the current menu selection state properly (maybe change background resource?)
- Update to use v7 version of ActionbarDrawerToggle (+Toolbar so that the navigation drawer slides over it)
- Animations for the menu toggling (when you show/hide the soldiers) ala I/O?

Commit:
Update LoginActivity to only store ONE user

Commit:
Disable interaction with soldier box when soldier count <= 1

Commit:
Changed from ActionBar --> Toolbar in conjunction with NavigationDrawer

karllindmark commented:
TODO:
- [x] Display relevant information on `HomeFragment`
- [x] Display the current selection state in the `NavigationDrawerFragment`
- [x] Animate the soldier toggling in the `NavigationDrawerFragment` (when clicking on the soldier box)?

Commit:
[MERGE] Merge Gradle changes

Commit:
Convert users --> soldiers

Commit:
Display information regarding soldier on HomeFragment (not user)

peter-budo commented on BF4Intel/src/main/java/com/ninetwozero/bf4intel/database/migrations/MigrationToOneZeroSix.java:
```diff
@@ -0,0 +1,33 @@
+    protected void doMigration(SQLiteDatabase db) {
+        dropTable(db, SummarizedSoldierStatsDAO.TABLE_NAME);
+        createNewSummarizedSoldierStatsTable(db);
+        // (SummarizedSoldierStatsDAO);
```
Comment, please remove

peter-budo commented on BF4Intel/src/main/res/layout/list_item_navdrawer_separator.xml:
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<View xmlns:android="http://schemas.android.com/apk/res/android"
+      android:layout_marginTop="7dp"
```
Hard coded

peter-budo commented on BF4Intel/src/main/res/layout/list_item_navdrawer_soldier.xml:
```diff
@@ -32,6 +24,20 @@
+
+  <ImageView
+    android:id="@+id/soldier_dropdown_arrow"
+    android:layout_width="32dp"
```
Hard coded

peter-budo commented on BF4Intel/src/main/res/layout/list_item_navdrawer_soldier_dropdown.xml:
```diff
@@ -32,6 +24,19 @@
+
+  <ImageView
+    android:id="@+id/soldier_dropdown_arrow"
+    android:layout_width="32dp"
```
Hardcoded

peter-budo commented on BF4Intel/src/main/res/values/content_strings.xml:
```diff
@@ -36,4 +36,5 @@
   <string name="cdesc_reward_image">Reward image</string>
   <string name="cdesc_completion_indicator">Completion indicator</string>
   <string name="cdesc_expansion_icon">Expansion icon</string>
+  <string name="cdesc_arrow">Arrow</string>
```
Arrow has no meaning to blind person "Back" has more meaning

karllindmark replied:
It's for the toggling purposes, so maybe "Toggle"?

Commit:
Update the selection state in the Navigation Drawer

Commit:
Cleanup for PR

Commit:
Add (currently empty) icons to the NavigationDrawer

peter-budo commented:
Two list items selected

Basically after fresh install, you search for soldier, soldier selected and you are redirected to Overview. Then you select any option from navigation draw. On next open of navigational drawer two list items are marked as selected.

This is because when search for soldier is done, user is redirected to overview, however `formatNavigationDrawerItem(View view, boolean selected)` is not called on `Home` to deselect and on `Overview` to select :wink: 

![device-2014-11-03-201923](https://cloud.githubusercontent.com/assets/512980/4889530/c2c3db04-6396-11e4-940b-a5d14c5d9bcb.png)

Commit:
Fix issue with NavigationDrawer selection

@peter-budo: Forgot to send it up last night though, d'oh.

Commit:
Move menu options into the NavigationDrawer

Commit:
Align NavigationDrawer separator to the design spec (in terms of color)

Commit:
Fix issue with missing ActionBar in the settings activity

peter-budo committed:
Added icons for navigation drawer

peter-budo committed:
Removed context menu icon from LoginActivity

peter-budo committed:
• Updates to changelog
 • Change version to to major instead of 1.0.6 to 1.1.0

Commit:
Minor touch-ups per G+ call

This also fixes the padding between elements, hooray!

...the menu is still a minigame though. Maybe put out as BETA release for the people
at G+ to try out? :-)

peter-budo committed:
Fixed back button problem

peter-budo commented:
The only two outstanding issues I can see is holo theme on dialog and navigation drawer icon position changing on selection/desection

karllindmark commented:
Last issue is only on 7''+?  Or sw720dp?
On Nov 11, 2014 9:57 AM, "Peter Mikloško" notifications@github.com wrote:

> The only two outstanding issues I can see is holo theme on dialog and
> navigation drawer icon position changing on selection/desection
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ninetwozero/com.ninetwozero.bf4intel/pull/173#issuecomment-62518961
> .

peter-budo committed:
Fixed selection update messing icon position

peter-budo committed:
Dialog header proposed solution. Implemented fo now on Assignments only

peter-budo committed:
• Applying dialog fragment custom title view across application

peter-budo commented:
@karllindmark ping me when you around

Problems that I noticed so far:
- [x] HDPI phone not showing navigation drawer image
- [x] HDPI phone & 10" tablet no navigation drawer left padding
- [x] Action bar back arrow navigation from details to parent activity messed up it goes to Overview, example, open Assignments, select an assignment to see. Once there press arrow on top and you end up in Overview. NOTE: Back button is fine, it is just AB arrow that is misleading

Commit:
Add missing curly brackets to the if

Commit:
Address issues with drawer position state

Commit:
Extract values

Commit:
Remove special spacing_* for normal-hdpi

Commit:
Fix issue with the padding being reset upon setting a background

peter-budo committed:
Navigation drawer background image provided in densities
